src/video.tsx

```diff
@@ -173,36 +182,46 @@ export class Video extends Component<VideoProps, VideoComponentState> {
   }
 
   private play = () => {
-    this.mediaRef.current && this.mediaRef.current.play();
+    this.currentElement && this.currentElement.play();
   }
 
   private pause = () => {
-    this.mediaRef.current && this.mediaRef.current.pause();
+    this.currentElement && this.currentElement.pause();
   }
 
   private navigate = (time: number) => {
-    this.setState({currentTime: time});
-    this.mediaRef.current && (this.mediaRef.current.currentTime = time);
+    this.setState({ currentTime: time });
+    this.currentElement && (this.currentElement.currentTime = time);
   }
 
   private setVolume = (volume: number) => {
-    this.setState({volume});
-    this.mediaRef.current && (this.mediaRef.current.volume = volume);
+    this.setState({ volume });
+    this.currentElement && (this.currentElement.volume = volume);
   }
 
   private setPlaybackSpeed = (playbackSpeed: number) => {
-    this.mediaRef.current && (this.mediaRef.current.playbackRate = playbackSpeed);
+    this.currentElement && (this.currentElement.playbackRate = playbackSpeed);
+  }
+
+  private get currentElement(): SourceElement | undefined {
+    if (this.videoRef.current) {
```

Comment:
Shouldn't we use the source type in order to return the current ref?

Reply:
done